### PR TITLE
Support Python3, use argparse

### PR DESCRIPTION
**Pull request by Hwesta**
* Switch to using argparse instead of docopt because argparse does input validation
* Support running with Python2 or Python3
* Add and update tests

### CODE REVIEW

File: requirements/local.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 -r base.txt
-ipython
+ipython==1.0.0
```
Comment (mistydemeo) [inline requirements/local.txt]: Why fix this to 1.0?

Comment (Hwesta) [inline requirements/local.txt]: To work with Python 3.2, which is what Ubuntu 12.04 ships for python 3.  "IPython supports Python 2.7 and 3.3 or newer. Our older 1.x series supports Python 2.6 and 3.2." :(

**Comment by mistydemeo** [comment]
:+1:

If this is being ported to Python 3, maybe use subprocess32 in place of subprocess on Python 2.x?

**Comment by Hwesta** [comment]
I'd like 3 to be the default, and 2 to be backwards compatibility. :) We don't use subprocess for much, so I think I'll leave it as is.  Thanks for letting me know about that library.

**Pull request merged**